### PR TITLE
Refactor of event page

### PR DESCRIPTION
**Pull request by smarcet**
Signed-off-by: smarcet <smarcet@gmail.com>

refactoring proposal

### CODE REVIEW

File: src/templates/event-page.js [hunk 1]
```diff
@@ -20,40 +20,27 @@ import AttendanceTrackerComponent from "../components/AttendanceTrackerComponent
 import { getEventById } from "../actions/event-actions";
 import { getDisqusSSO } from "../actions/user-actions";
 
-import { PHASES } from "../utils/phasesUtils";
-
 export const EventPageTemplate = class extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
```
Comment (santipalenque) [inline src/templates/event-page.js:27]: remove empty state

File: src/templates/event-page.js [hunk 2]
```diff
@@ -20,40 +20,27 @@ import AttendanceTrackerComponent from "../components/AttendanceTrackerComponent
...
-      firstRender: true,
     };
 
     this.onEventChange = this.onEventChange.bind(this);
```
Comment (santipalenque) [inline src/templates/event-page.js:30]: remove binding and use arrow functions

File: src/templates/event-page.js [hunk 3]
```diff
@@ -20,40 +20,27 @@ import AttendanceTrackerComponent from "../components/AttendanceTrackerComponent
...
+    this.props.getEventById(eventId);
   }
 
   componentDidUpdate(prevProps, prevState, snapshot) {
-    const { eventId, event } = this.props;
+    const { eventId } = this.props;
     const { eventId: prevEventId } = prevProps;
-
-    if (eventId !== prevEventId || !event || event.id !== eventId) {
+    if (eventId !== prevEventId) {
```
Comment (santipalenque) [inline src/templates/event-page.js:43]: if event.id != eventId we should call getEvent

File: src/templates/event-page.js [hunk 4]
```diff
@@ -69,53 +56,39 @@ export const EventPageTemplate = class extends React.Component {
...
 
-    const currentEvent = eventsPhases.find((e) => `${e.id}` === eventId);
-    const eventStarted = currentEvent && currentEvent.phase !== null ? currentEvent.phase : null;
-    const firstHalf = currentEvent?.phase === 0 ? nowUtc < (event?.start_date + event?.end_date) / 2 : null;
+    const eventStarted = event && event.start_date > nowUtc;
+    const firstHalf =  nowUtc < (event?.start_date + event?.end_date) / 2;
+    const isOngoing = event && event.start_date > nowUtc && nowUtc > event.end_date;
 
-    if (!firstRender && !loading && !event) {
+    console.log(`EventPageTemplate::render loading ${loading} eventStarted ${eventStarted} isOngoing ${isOngoing} event ${event.id}`);
```
Comment (santipalenque) [inline src/templates/event-page.js:66]: remove log

File: src/templates/event-page.js [hunk 5]
```diff
@@ -69,53 +56,39 @@ export const EventPageTemplate = class extends React.Component {
...
+
+    if (!event) {
```
Comment (santipalenque) [inline src/templates/event-page.js:68]: we should check loading also, if not it will flash this while fetching the event

Comment (smarcet) [reply src/templates/event-page.js:68]: done

File: src/templates/event-page.js (hunk 4 above)
Comment (smarcet) [reply src/templates/event-page.js:66]: done

File: src/templates/event-page.js (hunk 3 above)
Comment (smarcet) [reply src/templates/event-page.js:43]: done

File: src/templates/event-page.js (hunk 2 above)
Comment (smarcet) [reply src/templates/event-page.js:30]: done

File: src/templates/event-page.js (hunk 1 above)
Comment (smarcet) [reply src/templates/event-page.js:27]: done

**Pull request merged**